### ISSUE TITLE
AWS S3 Bucket Policy Get/Put Issues

### ISSUE THREAD

**Issue by dennykim** [aws/aws-sdk-ios#340]
I'm trying to use `AWSS3Service` `getBucketPolicy` and `putBucketPolicy` methods, but I'm having some difficulties. I've noticed a couple problems that may be related to each other:

1. The `policy` for `AWSS3GetBucketPolicyOutput` is defined as `NSString`, but the debugger seems to report it as `NSDictionary`.
![screen shot 2016-03-18 at 12 47 12 am](https://cloud.githubusercontent.com/assets/1584690/13868786/0d0a04a4-eca3-11e5-872e-60ecffd99da8.png)

2. The response from `getBucketPolicy` (`AWSS3GetBucketPolicyOutput.policy`) always seems to be empty (`NSDictionary`).

3. The `putBucketPolicy` is returning a JSON parse error and I cannot determine the root cause of it. I've tried different JSON document structures such as:
  ```
  {
     "Policy" : {...}
  }
  ```
  ```
  {
     "Id" : "...", 
     "Version" : "...",
     "Statement" : [...]
  }
  ```
  ```
  { 
     "Sid" : "...",
     "Action" : [...],
     "Effect" : "...",
     "Resource" : "...",
     "Principal" : "..."
  }
  ```
  Regardless of the keys, the error always reports a JSON parse error at the first character after the `{` and `"`. I've looked at some documentations for the structure for the JSON document, but none of them seem to work.

I would appreciate it if someone can investigate these issues.
Thanks!

**Comment by yosuke-matsuda** [comment]
Thank you for bringing the issue to our attention. It seems like the SDK escapes the `"` in the policy, and it is causing the validation error on the service side. We will take a look into this. Thanks.

**Comment by karthiksaligrama** [comment]
fixed with 2.4.8
